libtransmission/copy-test.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * This file is copyright Dr Robert Harvey Crowston, 2019.
+ * <crowston@protonmail.com>
+ *
+ * This file may be used and distributed under the GNU GPL versions 2 or 3 or
+ * any future version of the GPL.
+ *
+ */
+
+#include <string.h> /* memcmp() */
+
+#include "transmission.h"
+#include "crypto-utils.h"
+#include "file.h"
+#include "tr-assert.h"
+
+#include "libtransmission-test.h"
+
+/***
+****
+***/
+
+#define RANDOM_FILE_LENGTH  (1024 * 1024 * 128) /* 128 MiB */
```

Comment:
2-3 MiB would probably be enough to test the logic around 1 MiB buffer in POSIX implementation, maybe adding a few more bytes on top to check non-full buffer use. Is there a particular reason for the test file to be that big?

Reply:
I suppose I didn't think of 128 MB as a large file; typically I'm dealing with files in the 100s of GB. But I see your point; 128 MB doesn't really test anything that 4 MB wouldn't.